glotaran/builtin/megacomplexes/decay/decay_megacomplex_base.py

```diff
@@ -0,0 +1,117 @@
+"""This package contains the decay megacomplex item."""
+from __future__ import annotations
+
+import numpy as np
+import xarray as xr
+
+from glotaran.builtin.megacomplexes.decay.irf import IrfMultiGaussian
+from glotaran.builtin.megacomplexes.decay.k_matrix import KMatrix
+from glotaran.builtin.megacomplexes.decay.util import decay_matrix_implementation
+from glotaran.builtin.megacomplexes.decay.util import retrieve_decay_associated_data
+from glotaran.builtin.megacomplexes.decay.util import retrieve_irf
+from glotaran.builtin.megacomplexes.decay.util import retrieve_species_associated_data
+from glotaran.model import DatasetModel
+from glotaran.model import Megacomplex
+
+
+class DecayMegacomplexBase(Megacomplex):
+    """A Megacomplex with one or more K-Matrices."""
+
+    def get_compartments(self, dataset_model: DatasetModel) -> list[str]:
+        raise NotImplementedError
+
+    def get_initial_concentration(self, dataset_model: DatasetModel) -> np.ndarray:
+        raise NotImplementedError
+
+    def get_k_matrix(self) -> KMatrix:
+        raise NotImplementedError
```

Comment:
On second thought, I want to avoid using `ABC` here. The problem to solve here is to small for this, so I decided to remove subclassing at all and rather factoring out the shared methods to standalone functions.

This should also make it more clear that that decay megacomplexes are not intended to for general subclassing but that we only have 2 more specialized variants.